src/rdkafka_msgset_reader.c

```diff
@@ -1216,3 +1401,230 @@ rd_kafka_msgset_parse (rd_kafka_buf_t *rkbuf,
         return err;
 
 }
+
+
+/**
+ * @brief Offset comparator
+ */
+static int rd_kafka_offset_cmp (const void *_a, const void *_b) {
+        const int64_t *a = _a, *b = _b;
+        return *a - *b;
+}
+
+/**
+ * @brief Pid comparator for rd_kafka_aborted_txn_start_offsets_t
+ */
+static int rd_kafka_aborted_txn_cmp_by_pid (const void *_a, const void *_b) {
+        const rd_kafka_aborted_txn_start_offsets_t *a = _a, *b = _b;
+        return a->pid - b->pid;
+}
+
+
+/**
+ * @brief Free resources associated with an AVL tree node.
+ */
+void rd_kafka_aborted_txn_node_destroy (void *offsets) {
+        rd_list_destroy(&((rd_kafka_aborted_txn_start_offsets_t *)offsets)->offsets);
+}
+
+
+rd_kafka_aborted_txns_t *
+rd_kafka_aborted_txns_new (int32_t txn_cnt) {
```

Comment:
docstring

Reply:
this function is documented in the header. i think bad to duplicate that, and better this be in the header. the function above is an implementation detail, so documented in the .c.

Reply:
ahh, i see now that's going against the convention. will change. seems counter intuitive to put documentation in the bit that's supposed to be private and not put it in the bit that's supposed to be public though.

Reply:
So for the internal API what I do is document the function at its definition (in the .c file), close to the implementation to make sure the docs stay up to date,
while in the public API the functions are documented in the public rdkafka.h file (because that's what people see).